src/algebra/lie/basic.lean

```diff
@@ -474,6 +480,33 @@ lie_module.is_trivial.trivial x m
 abbreviation is_lie_abelian (L : Type v) [has_bracket L L] [has_zero L] : Prop :=
 lie_module.is_trivial L L
 
+lemma function.injective.is_lie_abelian {R : Type u} {L₁ : Type v} {L₂ : Type w}
+  [comm_ring R] [lie_ring L₁] [lie_ring L₂] [lie_algebra R L₁] [lie_algebra R L₂]
+  {f : L₁ →ₗ⁅R⁆ L₂} (h₁ : function.injective f) (h₂ : is_lie_abelian L₂) :
+  is_lie_abelian L₁ :=
+{ trivial := λ x y,
+    by { apply h₁, rw [lie_algebra.map_lie, trivial_lie_zero, lie_algebra.map_zero], } }
+
+lemma function.surjective.is_lie_abelian {R : Type u} {L₁ : Type v} {L₂ : Type w}
+  [comm_ring R] [lie_ring L₁] [lie_ring L₂] [lie_algebra R L₁] [lie_algebra R L₂]
+  {f : L₁ →ₗ⁅R⁆ L₂} (h₁ : function.surjective f) (h₂ : is_lie_abelian L₁) :
+  is_lie_abelian L₂ :=
+{ trivial := λ x y,
+    begin
+      obtain ⟨u, hu⟩ := h₁ x, rw ← hu,
+      obtain ⟨v, hv⟩ := h₁ y, rw ← hv,
+      rw [← lie_algebra.map_lie, trivial_lie_zero, lie_algebra.map_zero],
+    end }
+
+lemma lie_abelian_iff_equiv_lie_abelian {R : Type u} {L₁ : Type v} {L₂ : Type w}
+  [comm_ring R] [lie_ring L₁] [lie_ring L₂] [lie_algebra R L₁] [lie_algebra R L₂]
+  (e : L₁ ≃ₗ⁅R⁆ L₂) : is_lie_abelian L₁ ↔ is_lie_abelian L₂ :=
+begin
+  split; intros h,
+  { exact e.symm.injective.is_lie_abelian h, },
+  { exact @function.injective.is_lie_abelian R L₁ L₂ _ _ _ _ _ e e.injective h, },
```

Comment:
I don't understand why the following doesn't work instead:
```lean
exact e.injective.is_lie_abelian h,
```
Bizarrely, the following ridiculous code works though:
```lean
exact e.symm.symm.injective.is_lie_abelian h,
```
I will investigate if I have time this evening.

Reply:
I just pushed [a commit](https://github.com/leanprover-community/mathlib/pull/5930/commits/9e881ed336a5e7c05e1426c4cc3e4495b684e5bd) which I claim resolves this issue.